src/hooks/useMenu.ts

```diff
@@ -37,23 +37,17 @@ const menuReducer = (state: IMenuState, action: IMenuActions) => {
 
   switch (action.type) {
 
-    // initial State based in props and desktop size
+    // initial State based in props and desktop size and local storage
     case 'SET_MENU': {
-      const openMenu = localStorage.getItem(window.location.hostname + '_isMenuOpen');
-      let isMenuOpen = openMenu === 'true';
-      let isMenuPinned = openMenu === 'true';
-      let canPin = false;
+      const openValueStorage = localStorage.getItem(window.location.hostname + '_isMenuOpen');
 
-      if (action.data.canAlwaysPin || (action.data.defaultMenuOpen && action.data.desktopSize === 'xlarge')) {
-        canPin = true;
-      }
+      let isMenuOpen = (openValueStorage === 'true') || (openValueStorage === null && (!!action.data.defaultMenuOpen));
+      let isMenuPinned = (openValueStorage === 'true') && !!action.data.canAlwaysPin;
+      let canPin = (action.data.desktopSize === 'xlarge') || !!action.data.canAlwaysPin;
```

Comment:
canPin is not edited further bellow, please change this from let variable to a const.

Reply:
Done